scheduler/reconcile_util.go

```diff
@@ -321,8 +321,15 @@ func shouldFilter(alloc *structs.Allocation, isBatch bool) (untainted, ignore bo
 // updateByReschedulable is a helper method that encapsulates logic for whether a failed allocation
 // should be rescheduled now, later or left in the untainted set
 func updateByReschedulable(alloc *structs.Allocation, now time.Time, evalID string) (rescheduleNow, rescheduleLater bool, rescheduleTime time.Time) {
-	rescheduleTime, eligible := alloc.NextRescheduleTime()
+
+	// If the allocation is part of a deployment, only allow it to reschedule if
+	// it has been marked eligible for it explicitly.
+	if alloc.DeploymentID != "" && !alloc.DesiredTransition.ShouldReschedule() {
```

Comment:
If the alloc does not have a desired transition then alloc.DesiredTransition.ShouldReschedule() is false and this returns. Isn't that going to cause problems for failed allocs that don't have an update stanza and are not managed by the deployment watcher